service/src/main/java/org/cbioportal/service/impl/ExpressionEnrichmentServiceImpl.java

```diff
@@ -131,15 +131,21 @@
                 .map(ExpressionEnrichment::getEntrezGeneId)
                 .collect(Collectors.toList());
 
-        Map<Integer, List<ReferenceGenomeGene>> geneMapByEntrezId = refGeneService
-                .fetchGenesByGenomeName(entrezGeneIds, molecularProfile.getCancerStudy().getReferenceGenome())
+        Map<Integer, List<Gene>> geneMapByEntrezId = geneService
+                .fetchGenes(entrezGeneIds
+                    .stream()
+                    .map(Object::toString)
+                    .collect(Collectors.toList()),
+            "ENTREZ_GENE_ID",
+            "SUMMARY")
                 .stream()
-                .collect(Collectors.groupingBy(ReferenceGenomeGene::getEntrezGeneId));
+                .collect(Collectors.groupingBy(Gene::getEntrezGeneId));
 
         expressionEnrichments.forEach(expressionEnrichment -> {
-            ReferenceGenomeGene gene = geneMapByEntrezId.get(expressionEnrichment.getEntrezGeneId()).get(0);
-            expressionEnrichment.setHugoGeneSymbol(gene.getHugoGeneSymbol());
-            expressionEnrichment.setCytoband(gene.getCytoband());
+            Gene gene = geneMapByEntrezId.get(expressionEnrichment.getEntrezGeneId()).get(0);
+            if (gene != null) {
```

Comment:
I think gene won't be null any time, because only validated data would be loaded in to database

Reply:
if its null then you should remove that particular `expressionEnrichment` object from response